### PR TITLE
tests: add a test to verify UEFI boot on x86_64 targets.

### PR DESCRIPTION
**Pull request by chen-png**
add a new test to verify UEFT bootable method on x86_64
targets, it also could be tested on qemu_x86_64 platforms
based on uefi-run and OVMF.

Signed-off-by: Chen Peng1 <peng1.chen@intel.com>

### CODE REVIEW

**Comment by chen-png** [comment]
hi @nashif Could you help take a look of this? I added a test to verify the UEFI boot on qemu_x86_64 platform. 
the only failure in CI pipeline is that for the new test, it needs external uefi-run and OVFM environment.

**Comment by nashif** [comment]
> hi @nashif Could you help take a look of this? I added a test to verify the UEFI boot on qemu_x86_64 platform.
> the only failure in CI pipeline is that for the new test, it needs external uefi-run and OVFM environment.

right, those need to be installed and available in the CI environment

**Comment by chen-png** [comment]
@nashif ok, got it, thanks. I found CI environment used the docker image zephyrprojectrtos/ci:v0.17.1, I will add uefi-run and OVMF into the latest docker file.

**Comment by chen-png** [comment]
added uefi-run and ovmf into CI docker in https://github.com/zephyrproject-rtos/docker-image/pull/68

File: tests/misc/test_uefi_boot/README.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+
+Requirements
+############
+
+Please install uefi-run in your system environment according to this
+reference link https://github.com/Richard-W/uefi-run.
+
+Please install OVMF in your system environment according to this
+reference link https://github.com/tianocore/tianocore.github.io/wiki/OVMF.
```
Comment (andyross) [inline tests/misc/test_uefi_boot/README.rst:14]: Are there matching PRs to add these as Zephyr modules?  Having manually-installed external dependencies is sort of OK for an oddball board, but this is a CI platform.  Seems like this shouldn't be too hard.

File: tests/misc/test_uefi_boot/README.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
+UEFI Boot
+###########
```
Comment (andyross) [inline tests/misc/test_uefi_boot/README.rst:2]: I'd put this documentation in the platform layer somewhere.  This test isn't the only place where one might want to build EFI code under qemu.

**Review by andyross — CHANGES_REQUESTED** [review]
Love the integration work and the docs.  I just don't think this should be a new test if all it is doing is the equivalent of "Hello World!".  How about moving the OVMF integration into the qemu board configuration and just adding a new test case to samples/hello_world to turn on that one kconfig?

**Comment by chen-png** [comment]
> moving the OVMF integration into the qemu board configuration

yes, it's better, I will update it.

File: tests/misc/test_uefi_boot/README.rst (hunk 1 above)
Comment (chen-png) [reply tests/misc/test_uefi_boot/README.rst:14]: @andyross  for CI, I have a PR zephyrproject-rtos/docker-image#68 to add those dependencies into CI docker image, then CI could run this test, for developers, I think they should install it by themselves.

for running this test, we only need final binary uefi-run and ovmf.fd, so do we need to add them as a zephyr module and build them from source code? actually, I didn't build them from source code, I downloaded the release binary files directly.

**Comment by chen-png** [comment]
@andyross Could you help review it again? I made some changes according to your comments, thanks.

**Review by andyross — APPROVED** [review]
I still think adding these as modules is preferable to doing it in CI only; modules can be made to work automatically for everyone.  But having the patch in CI gets us test coverage so surely that's good enough.

**Review by LeiW000 — APPROVED** [review]

**Pull request merged**